projects/demo-maps/src/app/route-components/route-example-olperformance/fetchbutton/fetchbutton.component.ts

```diff
@@ -0,0 +1,113 @@
+import { Component, OnInit } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { LayersService, VectorLayer } from '@dlr-eoc/services-layers';
+import { Fill as olFill, Stroke as olStroke, Style as olStyle } from 'ol/style';
+import simplify from '@turf/simplify';
+import bbox from '@turf/bbox';
+import { reproject } from 'reproject';
+import proj4 from 'proj4';
+import { BehaviorSubject } from 'rxjs';
+
+
+type fetchState = 'ready' | 'fetching' | 'done' | 'error';
+
+
+@Component({
+  selector: 'app-fetchbutton',
+  templateUrl: './fetchbutton.component.html',
+  styleUrls: ['./fetchbutton.component.scss']
+})
+export class FetchbuttonComponent implements OnInit {
+
+  public state$ = new BehaviorSubject<fetchState>('ready');
+
+  constructor(
+    private http: HttpClient,
+    private layersSvc: LayersService
+  ) { }
+
+  ngOnInit(): void {
+  }
+
+  fetchClicked() {
+    const fileUrl = 'https://bloomington.in.gov/geoserver/publicgis/ows?service=WFS&version=1.0.0&request=GetFeature&typeName=publicgis:Buildings&outputFormat=application%2Fjson';
+    this.state$.next('fetching');
+    this.http.get(fileUrl).subscribe((data) => {
+      this.addLayers(data);
+      this.state$.next('done');
+    });
+  }
+
+  private addLayers(dataOrig: any) {
```

Comment:
please try to use separation of concerns and don't use components to create the map data.
The UKIS design is to crete map layers and and data only in the route/view component and if it gets to messy set up simple helper calsses like map.utils.ts or map.layers.ts